### PR TITLE
Update isomorphic routing docs

### PR DESCRIPTION
**Pull request by noahtallen**
#### Proposed Changes
Updates the isomorphic routing docs with more information about logged-in requests and caching.

### CODE REVIEW

File: docs/isomorphic-routing.md [hunk 1]
```diff
@@ -48,6 +48,30 @@ as that has a lot of dependencies that aren't ready for server-side rendering.
   our server router's middleware signatures. We might want to switch to an isomorphic
   router in the future.
 
+## Logged-in requests
+Currently, if a request is not logged-in, SSR middleware chains are skipped.
```
Comment (lynnbgriffith) [inline docs/isomorphic-routing.md]: I think you intended “if a request is logged-in” here.

File: docs/isomorphic-routing.md [hunk 2]
```diff
@@ -48,6 +48,30 @@ as that has a lot of dependencies that aren't ready for server-side rendering.
...
+Even if you define an index.node.js, those middleware handlers will not be used
+if the request is logged in. This improves server-side performance in scenarios
```
Comment (lynnbgriffith) [inline docs/isomorphic-routing.md]: Let’s use “logged-in” here just to be consistent throughout.

File: docs/isomorphic-routing.md [hunk 3]
```diff
@@ -48,6 +48,30 @@ as that has a lot of dependencies that aren't ready for server-side rendering.
...
+where the SSR pipeline ultimately resolves the same thing the non-SSR code would
+resolve.
+
+In general, SSR isn't a good fit for our logged-in pages. A lot of data is different
+for different users, which makes it difficult to correctly cache data for the same
+page. Since caching data is crucial for performance, it isn't usually feasable to
```
Comment (lynnbgriffith) [inline docs/isomorphic-routing.md]: s/feasable/feasible/

File: docs/isomorphic-routing.md [hunk 4]
```diff
@@ -48,6 +48,30 @@ as that has a lot of dependencies that aren't ready for server-side rendering.
...
+## Network requests
+It's common for SSR layouts to include network-fetched data. For example, the themes
+SSR page includes themes fetched on the backend. If your SSR section includes
+network requsets, it is **crucial** for those requsets to be cached and optimized
+as much as possible. Caching cannot be a follow-up enhancement -- without caching,
+performance across the server can be negatively impacted, resulting in route timeouts
+and even outages.
+
+Thankfully, we have existing techniques for caching data, such as the redux cache
+(used for themes) and the react query cache (used for plugins). In each case, you
```
Comment (lynnbgriffith) [inline docs/isomorphic-routing.md]: Either react-query or React Query. Although, I suppose it’s now technically “TanStack Query”.

**Review by lynnbgriffith — APPROVED** [review]
Suggested a few minor edits, but otherwise looks good. Thanks for updating the docs!

**Comment by noahtallen** [comment]
Nice catch, thanks for taking a look!

File: docs/isomorphic-routing.md (hunk 4 above)
Comment (noahtallen) [reply docs/isomorphic-routing.md]: I'll stick with React Query, given the org is most familiar with that for now :P

**Pull request merged**